src/index.js

```diff
@@ -134,6 +134,20 @@ export default class Carousel extends React.Component {
     }
   }
 
+  componentDidUpdate() {
+    this.controlsMap = [
```

Comment:
Can we collapse this to maybe a helper function since it's copy and paste from elsewhere in this file?

Also, can we get a regression unit test when you get a moment? Thanks!

@mhink -- If you've got time, can you chat with @zhaostan and maybe pick this up if he doesn't have time? Thanks!